pkg/virt-handler/vm.go

```diff
@@ -687,6 +687,46 @@ func (d *VirtualMachineController) migrationSourceUpdateVMIStatus(origVMI *v1.Vi
 	return nil
 }
 
+func findMultusIP(vmi *v1.VirtualMachineInstance, multusIp *string) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return
+	}
+	for _, iface := range ifaces {
+		if iface.Flags&net.FlagUp == 0 {
+			continue // interface down
+		}
+		if iface.Flags&net.FlagLoopback != 0 {
+			continue // loopback interface
+		}
+		addrs, err := iface.Addrs()
+		if err != nil {
+			return
+		}
+		for _, addr := range addrs {
+			var ip net.IP
+			if iface.Name != "net1" {
```

Comment:
Would it make sense to use the pod [downward api](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#capabilities-of-the-downward-api) to get `k8s.v1.cni.cncf.io/networks-status` annotation info into virt-handler and in addition a commandline flag for virt-handler where you pass in the multus network name?

One can then look up info like this:

```
    k8s.v1.cni.cncf.io/networks-status: |- 
      [{
          "name": "openshift-sdn",
          "interface": "eth0",
          "ips": [
              "10.128.2.14"
          ],
          "default": true,
          "dns": {}
      },{
          "name": "macvlan-bridge",
          "interface": "net1",
          "ips": [
              "20.2.2.100"
          ],
          "mac": "22:2f:60:a5:f8:00",
          "dns": {}
      }]
```

This would probably remove all guessing and you could simply read the IP from this info, since you would know the multus network name from the commandline flag.

Reply:
Force-pushed switch to downward API, let me know what you think!
Haven't done the commandline part yet. Would it make sense to parse the json part of the NetworkAttachmentDefinition to find the network name and use that as a key instead of the interface name?

Reply:
Yes, I think so. Just using the multus name as key and then extracting all info from there sounds great.

Reply:
So, I started working on that, using https://github.com/k8snetworkplumbingwg/network-attachment-definition-client to access the NetworkAttachmentDefinition.
However, the client above fails to compile. I'm getting obscure errors I wouldn't know how to fix. (Starts with: `vendor/github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/clientset/versioned/typed/k8s.cni.cncf.io/v1/k8s.cni.cncf.io_client.go:74:32: undefined: serializer.DirectCodecFactory`).
Even if it worked, finding the NetworkAttachmentDefinition and parsing the json blob it contains adds a lot of complxity to the code, so I'm not sure it's worth it...
Maybe we could request that the name of the network matches the name of the NetworkAttachmentDefinition? Or add fields in the CR? (we'll need a second one anyway if we want to allow using NetworkAttachmentDefinitions that are in a different namespace than `kubevirt`)
@rmohr please let me know what you think! Thanks

Reply:
@jean-edouard did you consider simply parsing the json into a `map[string]interface{}`? But in general you don't need the client. You only need the api definition to deserialize the json. Consider simply copying the `NetworkStatus` into our codebase (https://github.com/k8snetworkplumbingwg/network-attachment-definition-client/blob/master/pkg/apis/k8s.cni.cncf.io/v1/types.go#L101). It is nice if we can reuse external code, but at the end there is a `v1` REST API defined by it, which means that we can define our own struct for the API without risks.

Reply:
But looking at your latest code, you basically already did that. So maybe I did not understand what you try to do? I think in general you will not have to ever look up a `NetworkAttachmentDefinition` in the cluster if I did not miss anything.

Reply:
This is probably a lack of knowledge on my side, but is there a way to retrieve a NetworkAttachmentDefinition from k8s in golang without having to use a client like network-attachment-definition-client?
If so, could you please point me to some docs/example on how to do that?
Thanks!

Reply:
Ah, so maybe this is a misunderstanding. In the CR, the cluster admin provides a NetworkAttachmentDefinition by name.
Inside that object there is a json blob (don't ask me why), with a different name, and that's the name I need to parse the yaml network info!

Reply:
This was discussed offline, and Roman may have found a specific notation that could potentially alleviate this hole issue.
I'm looking into that now.

Reply:
Indeed, the network annotation can name the interface using the "@" notation. Pushed, thanks!